package/MDAnalysis/transformations/translate.py

```diff
@@ -138,4 +150,173 @@ def wrapped(ts):
+    
+    Example
+    -------
+    
+    Translate the center of mass of the second residue of the universe u to the center of the
+    xy plane:
+
+    .. code-block:: python
+    
+        ag = u.residues[1].atoms
+        transform = MDAnalysis.transformations.center_in_plane(ag, xy, center_of='mass')
+        u.trajectory.add_transformations(transform)
+    
+    Parameters
+    ----------
+    ag: AtomGroup
+        atom group to be centered on the unit cell.
+    plane: str
+        used to define the plane on which the given AtomGroup will be centered. Suported
+        planes are yz, xz and xy planes.
+    center_to: array-like or str
+        coordinate from which the axes are centered. Can be an array of three coordinate 
+        values or `center` which centers the AtomGroup coordinates on the center of the 
+        unit cell. Default is `center`.
+    center_of: str, optional
+        used to choose the method of centering on the given atom group. Can be 'geometry'
+        or 'mass'
+    wrap: bool, optional
+        If `True`, all the atoms from the given AtomGroup will be moved to the unit cell
+        before calculating the center of mass or geometry. Default is `False`, no changes
+        to the atom coordinates are done before calculating the center of the AtomGroup. 
+    
+    Returns
+    -------
+    MDAnalysis.coordinates.base.Timestep
+    
+    """
+    
+    pbc_arg = wrap
+    if plane is not None:
+        if plane not in ('xy', 'yz', 'xz'):
+            raise ValueError('{} is not a valid plane'.format(plane))
+        axes = {'yz' : 0, 'xz' : 1, 'xy' : 2}
+        plane = axes[plane]
+    if center_to is not None:
+        if isinstance(center_to, string_types):
+            if not center_to == 'center':
+                raise ValueError('{} is not a valid "center_to"'.format(center_to))
+        else:
+            center_to = np.asarray(center_to, np.float32)
+            if center_to.shape != (3, ) and center_to.shape != (1, 3):
+                raise ValueError('{} is not a valid "center_to"'.format(center_to))
+    try:
+        if center_of == 'geometry':
+            center_method = partial(ag.center_of_geometry, pbc=pbc_arg)
+        elif center_of == 'mass':
+            center_method = partial(ag.center_of_mass, pbc=pbc_arg)
+        else:
+            raise ValueError('{} is not a valid argument for "center_of"'.format(center_of))
+    except AttributeError:
+        if center_of == 'mass':
+            raise AttributeError('{} is not an AtomGroup object with masses'.format(ag))
+        else:
+            raise ValueError('{} is not an AtomGroup object'.format(ag))
+
+    def wrapped(ts):
+        boxcenter = ts.triclinic_dimensions.sum(axis=0) / 2.0
+        position = boxcenter
+        if center_to != 'center':
+            _origin = center_to
+            position[plane] = _origin[plane]
+        vector = np.asarray(position, np.float32) - center_method()
```

Comment:
Are you really working in 2D there? From what I read you are centering in the box, not the plane.